pulsar-broker/src/test/java/org/apache/pulsar/client/api/TopicReaderTest.java

```diff
@@ -650,4 +647,44 @@ public void testReaderIsAbleToSeekWithTimeOnMiddleOfTopic() throws Exception {
         reader.close();
         producer.close();
     }
+
+    @Test
+    public void testReaderStartMessageIdAtExpectedPosition() throws Exception {
+        final int numOfMessage = 10;
+        final String topicName = "persistent://my-property/my-ns/ReaderStartMessageIdAtExpectedPosition";
+        final int resetIndex = new Random().nextInt(numOfMessage); // Choose some random index to reset
+
+        Producer<byte[]> producer = pulsarClient.newProducer()
+                .topic(topicName).create();
+
+        MessageId resetPos = null;
+        for (int i = 0; i < numOfMessage; i++) {
+
+            MessageId msgId = producer.send(String.format("msg num %d", i).getBytes());
+            if (resetIndex == i) {
+                resetPos = msgId;
+            }
+        }
+
+        Reader<byte[]> reader = pulsarClient.newReader()
+                .topic(topicName)
+                .startMessageId(resetPos)
```

Comment:
The javadoc says that the message will be the one *after* the resetPos. Typically, the "resetPos" is the messageId you checkpoint after processing. 

Indeed there might be cases in which one desires to start reading from that message included. This should be achieved through some other config option.

Reply:
> The javadoc says that the message will be the one after the resetPos

Sorry, didn't know that.

> This should be achieved through some other config option.

Do you want me to refactor this pull in order to support that ?

Maybe `startMessageIdInclusive()` ?

Reply:
Refactor done! please @merlimat @jiazhai take a look now.

If everything is ok, I will then add javadoc to the new method on ReaderBuilder.